src/AppBundle/Command/AmbassadorShiftTimeLogCommand.php

```diff
@@ -0,0 +1,91 @@
+<?php
+// src/AppBundle/Command/AmbassadorShiftTimeLogCommand.php
+namespace AppBundle\Command;
+
+use Swift_Message;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Doctrine\ORM\Query\Expr\Join;
+
+class AmbassadorShiftTimeLogCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+          ->setName('app:shift:send_late_shifters')
+          ->setDescription('Send shifters which are too late')
+          ->setHelp('This command allows you to send alerts for shifters that are too late')
+            ->addOption('emails', null, InputOption::VALUE_OPTIONAL, 'Email recipients (comma separated)')
+            ->addOption('emailTemplate', null, InputOption::VALUE_OPTIONAL, 'Template used in email alerts', 'SHIFT_LATE_ALERT_EMAIL');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $email_template = $input->getOption('emailTemplate');
+
+        $alerts = $this->computeAlerts();
+        $nbAlerts = count($alerts);
+        if ($nbAlerts > 0) {
+            $output->writeln('<fg=cyan;>Found ' . $nbAlerts . ' alerts to send</>');
+            $this->sendAlertsByEmail($input, $output, $alerts, $email_template);
+        } else {
+            $output->writeln('<fg=cyan;>No shift alert to send</>');
+        }
+    }
+
+    private function computeAlerts() {
+        $time_after_which_members_are_late_with_shifts = $this->getContainer()->getParameter('time_after_which_members_are_late_with_shifts');
+
+        $em = $this->getContainer()->get('doctrine')->getManager();
```

Comment:
envisageable de réutiliser `$formHelper->initSearchQuery($this->getDoctrine()->getManager());` ? je n'y connais pas assez, mais c'est dommage de réecrire des bouts de code qui sont déjà dans `SearchUserFormHelper` (sans parler du reste qui est aussi dans `AmbassadorController.shiftTimeLogAction`)

Reply:
Effectivement j’ai "pompé" ce bout de code à l’identique de `initSearchQuery`, parce que je n’étais pas dans un contexte de "formulaire" donc ça ne me semblait pas approprié d’utiliser ce module.

Mais si ça ne pose pas de problème je peux effectivement le réutiliser.

Je peux voir pour factoriser la partie commune avec `AmbassadorController.shiftTimeLogAction` si tu veux aussi, mais je ne suis pas certain de l’endroit approprié où le faire, donc si tu as une indication je la suivrai sinon je ferai au mieux :).

Reply:
je comprends complètement, j'aimerais bien factoriser un maximum, mais je ne sais pas non plus quelle est la meilleure facon de faire, je crois que c'est grâce aux Repository. si c'est trop compliqué on peut juste mettre un commentaire du genre `// similar to AmbassadorController.shiftTimeLogAction query`

Reply:
Ce n’est pas compliqué dans le sens technique (en tout cas c’est dans mes compétences), la difficulté de mon côté est que je n’ai pas l’habitude de symfony donc je ne connais pas les conventions associées. Je vais voir pour Repository, ça a l’air de correspondre à un bon endroit où définir ce genre de fonction ;)

Reply:
juste pour être sûr : tu veux faire la modif vers le Repository dans cette PR, ou tu veux faire la modif plus tard (et dans ce cas là on merge) ?

Reply:
Non non je prévois de le faire dans cette PR mais je dois prendre le temps de le faire. Mais si tu penses que c’est pressé d’avoir cette feature tu peux merger et j’en ouvrirai une autre

Reply:
pas du tout pressé de notre côté, c'est toi qui voit :)